src/sources/util/tcp.rs

```diff
@@ -53,17 +53,17 @@ async fn make_listener(
     }
 }
 pub trait IsErrorFatal {
-    fn is_error_fatal() -> bool;
+    fn is_error_fatal(&self) -> bool;
```

Comment:
Started injecting `self` here as the `fluent` source decides which errors are fatal based on whether they are i/o or decode errors (only i/o errors are fatal).